DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestForControllers/TestGitHubExtensionSecurityWebApiAccountControllerAssignRolesToUser.cs

```diff
@@ -0,0 +1,181 @@
+            var userManager = Substitute.For<ApplicationUserManager>(store);
+            List<User> users = new List<User>{
+                new User{ProviderId = 1,},
+                new User{ProviderId =2, },
+                new User{ProviderId=3,},
+            };
+
+            userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            var context = Substitute.For<ISecurityContext>();
+            IEnumerable<SecurityRole> roles = new List<SecurityRole>
+            {
+                new SecurityRole{ Id = 1, Name = "Admin" },
+                new SecurityRole{ Id = 1, Name = "Developer" },
+                new SecurityRole{ Id = 1, Name = "Reviewer" }
+            };
+
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+
+            
+
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                context, Substitute.For<IAuthService>(), userManager, store,
+                Substitute.For<IRoleStore<IdentityRole, string>>());
+
+            IHttpActionResult nullUser = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign).Result;
+
+            Assert.IsType<InvalidModelStateResult>(nullUser);
+        }
+
+        [Fact]
+        public void CheckTheErrorMessageIfRoleNotExists()
+        {
+            int gitHubId = 1;
+            int repoId = 0;
+            string roleToAssign = "BussinessAnalytics";
+            string expectedError = string.Format("Roles '{0}' does not exists in the system", roleToAssign);
+            var store = Substitute.For<IUserStore<User>>();
+            var userManager = Substitute.For<ApplicationUserManager>(store);
+            List<User> users = new List<User>{
+                new User{ProviderId = 1,},
+                new User{ProviderId =2, },
+                new User{ProviderId=3,},
+            };
+
+            userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            var context = Substitute.For<ISecurityContext>();
+            IEnumerable<SecurityRole> roles = new List<SecurityRole>
+            {
+                new SecurityRole{ Id = 1, Name = "Admin" },
+                new SecurityRole{ Id = 1, Name = "Developer" },
+                new SecurityRole{ Id = 1, Name = "Reviewer" }
+            };
+
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+
+
+
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                context, Substitute.For<IAuthService>(), userManager, store,
+                Substitute.For<IRoleStore<IdentityRole, string>>());
+
+            IHttpActionResult nullUser = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign).Result;
+
+            InvalidModelStateResult modelStateResult = Assert.IsType<InvalidModelStateResult>(nullUser);
+            Assert.Equal(expectedError,modelStateResult.ModelState["role"].Errors.First().ErrorMessage);
```

Comment:
Is't not good practice to verify value by "magic" string. It should be moved to constant or transform to typed value. It's simplify next code refactoring.